pkg/services/ngalert/store/alert_rule.go

```diff
@@ -156,7 +158,7 @@ func (st DBstore) UpsertAlertRules(rules []UpsertRule) error {
 				existingAlertRule, err := getAlertRuleByUID(sess, r.New.UID, r.New.OrgID)
 				if err != nil {
 					if errors.Is(err, ngmodels.ErrAlertRuleNotFound) {
-						return nil
+						return errors.Wrapf(err, "failed to get alert rule: %s", r.New.UID)
```

Comment:
Let's leave this as-is for now and add it as a goal to remove as soon as we have test coverage across an API level. I've created a follow up for this - as we're also using it in several other parts of the alerting codebase.